### DISCUSSION TITLE
Is it incorrect to re-use guards by parametizing them?

### DISCUSSION THREAD

**Discussion by christopher-francisco-jpmc** [statelyai/xstate#4919] [Q&A] [answered]
I'm new to xstate and refactoring an existing machine. For simplicity, I've got something like this:

```
always: [
  { target: 'EAT',      cond: { type: 'isDoing', expected: 'hungry', actual: currentBehaviour },
  { target: 'SLEEP',  cond: { type: 'isDoing', expected: 'tired',     actual: currentBehaviour },
  // ...repeat for a total of 14 of these
]

...

guards: {
  isDoing: (_, _, { cond: { expected, actual } ) => expected === actual,
```

I haven't actually ran it as there are some other refactors in place; but the VSCode plugin is properly able to draw the diagram, so I'm assuming it runs. 

The problem is, the VSCode plugin is showing a few orange question marks that say: `Found 14 guards with the name 'isDoing'` which leads me to believe that maybe I'm not supposed to be using the same guard multiple times like this.

I would appreciate some pointers, am I doing it wrong?

**Comment by davidkpiano** [comment]
Which version of XState are you using? It looks like you're using v4. This might be a problem with the extension.

But yes, in v5, you can use parameters and it's actually encouraged. See here: https://stately.ai/docs/guards#guard-object

**Reply by christopher-francisco-jpmc** [reply]
Yes this is v4, apologies I meant to include that info, but it slipped.

Should I?
  * ignore it
  * use guards differently
  * look for a potential different problem that may be causing this warning/error to show

We will update to v5 in a few months but for the time being I have to focus in v4

**Reply by davidkpiano** [reply, marked as answer]
You can use parameterized guards in v4 as well. If you see any warning/error, that's probably a problem with the extension.
